layers/LAYERFORMATv4_2.md

```diff
@@ -0,0 +1,185 @@
+
+## Version Object Properties
+
+| Name     | Type     | Required? | Default Value (if not present) | Description |
+| :------------- | :------------- | :------------- | :------------- | :------------- |
+| attack | String | No | Current version of ATT&CK: "8" | ATT&CK version of this layer |
+| navigator | String | Yes | | Must be "4.2" |
+| layer | String | Yes | | Must be "4.1" |
+
+## Technique Object properties
+
+Technique objects are used to store both techniques and subtechniques. The only difference in representation between a technique and a subtechnique is in the techniqueID field, which for subtechniques is the parent technique ID followed by the subtechnique-id suffix.
+
+| Name     | Type     | Required? | Default Value (if not present) | Description |
+| :------------- | :------------- | :------------- | :------------- | :------------- |
+| techniqueID | String | Yes | n/a | Unique identifier of the ATT&CK technique, e.g. "T####". For subtechniques, the format is "T####.###", where the substring to the left of the decimal is the parent technique ID, and the right-side substring is the subtechnique ID suffix. |
+| tactic | String | No | n/a | Unique identifier of the ATT&CK technique's tactic, e.g. "lateral-movement". If the field is not present, the annotations for the technique will appear under every tactic the technique belongs to |
+| comment | String | No | "" | Free-text field |
+| enabled | Boolean | No | true | Specifies if the technique is considered enabled or disabled in this layer |
+| score | Number | No | (unscored) | Optional numeric score assigned to this technique in the layer. If omitted, the technique is considered to be "unscored" meaning that it will not be assigned a color from the gradient by the Navigator |
+| color | String | No | "" | Explicit color value assigned to the technique in this layer. Note that explicitly defined color overrides any color implied by the score – the Navigator will display the technique using the explicitly defined color |
+| metadata | Array of Metadata objects and Metadata Separator objects | No | | User defined metadata for this technique. See definition of Metadata object and Metadata Separator object below |
+| showSubtechniques | boolean | No | false | if true, the sub-techniques under this technique will be shown by default. This field is only valid under a technique with subtechniques. Note that subtechniques can still be shown/hidden using the UI controls — this field is simply the default state. |
```

Comment:
I was thinking about not including the aggregateScore into the layer json, because it bloats the json file with the aggregate score of all techniques, when the aggregateScore can be calculated on render (if the user uploads a json file with showAggregateScores set to true). 

What do you think? 

If that's okay, then I'll make the changes to the serialize and deserialize methods.

Reply:
Yeah I think it's fine to remove. I have another comment somewhere on this PR about possible bugs from including it, and it's pretty straightforward for users to calculate it themselves for anything built to ingest layers. So I'm all for removing it from layers.